base/sysinfo.jl

```diff
@@ -213,6 +252,23 @@ function cpu_summary(io::IO=stdout, cpu::AbstractVector{CPUinfo} = cpu_info())
     _cpu_summary(io, cpu, first, length(cpu))
 end
 
+"""
+    Sys.cpu_info()
+
+Return a vector of `CPUinfo` objects, where each object represents information about a CPU core.
+This is displayed in a tabular format by the REPL, so most users will not need to access the `CPUinfo`
+data structures directly.
+
+The function provides information about each CPU, including model, speed, and usage statistics such as user time, nice time, system time, idle time, and interrupt time.
+The `CPUinfo` type is a mutable struct with the following fields:
```

Comment:
As commented earlier, if we are going to mention the `CPUinfo` type in the docstrings, then it becomes part of the public API, and there should at least be a
```jl
public CPUinfo
```
statement reflecting this, as well as moving this information about the `struct` into a docstring for `CPUinfo`.